Resolve CI issues (peer dependencies, version support, end-to-end testing)

dopry opened the pull request:
npm WARN @angular-devkit/schematics@0.0.52 requires a peer of @angular-devkit/core@0.0.29 but none is installed.
npm WARN @schematics/angular@0.1.17 requires a peer of @angular-devkit/core@0.0.29 but none is installed.

closes #1

dopry committed:
chore: resolve unmet peer dependeny warnings

npm WARN @angular-devkit/schematics@0.0.52 requires a peer of @angular-devkit/core@0.0.29 but none is installed.
npm WARN @schematics/angular@0.1.17 requires a peer of @angular-devkit/core@0.0.29 but none is installed.

bufferoverflow commented:
@dopry You should adopt the commit message, to e.g. `chore: resolve unmet dependencies` as conventional commit messages are used here. you can verify your message with `npm run changelog`, see https://github.com/marionebl/commitlint/tree/master/%40commitlint/config-conventional

dopry commented:
@bufferoverflow, thanks. done.

dopry committed:
fix(ci): pass nosandbox when running headless for travis-ci

see: https://docs.travis-ci.com/user/chrome

dopry committed:
fix(ci): update node.js support to >= 6.4

There are testing failures in 4.x and 5.x from
syntax issues in angular. 6.x was chosen as a
base since ES6 is on the roadmap for bacstack and
ES6 only has a 57% complete implementination

dopry committed:
fix(ci): streamline karma for ci

* single run
* no autoWatch
* no html reports
* no html coverage reports

dopry committed:
chore: fast fail builds, allow 9.x to fails without failing whole matrix

dopry committed:
chore: update karma to use electron instead of chrome

dopry committed:
fix(tests): inject dependencies in tests

dopry committed:
fix(ci): add lcov reporting for coveralls
